src/utils/schemas.js

```diff
@@ -19,3 +24,103 @@ export function getSchemaByClassName(className) {
+
+export function typeofSchema(schema) {
+    if (has(schema, "type")) {
+        return schema.type;
+    }
+    if (has(schema, "properties")) {
+        return "object";
+    }
+    if (has(schema, "items")) {
+        return "array";
+    }
+}
+
+function getEnumValues(nodes) {
+    if (!nodes.length) return {};
+    return {
+        enum: nodes.map((node) => get(node, node.dataSelector.value)),
+    };
+}
+
+function getEnumNames(nodes) {
+    if (!nodes.length) return {};
+    return {
+        enumNames: nodes.map((node) => get(node, node.dataSelector.name)),
+    };
+}
+
+/**
+ * @summary Recursively generate `dependencies` for RJSF schema based on tree.
+ * @param {Object[]} nodes - Array of nodes (e.g. `[tree]` or `node.children`)
+ * @returns {{}|{dependencies: {}}}
+ */
+export function buildDependencies(nodes) {
+    if (nodes.length === 0 || nodes.every((n) => !n.children?.length)) return {};
+    const parentKey = nodes[0].dataSelector.key;
+    const childKey = nodes[0].children[0].dataSelector.key;
+    return {
+        dependencies: {
+            [parentKey]: {
+                oneOf: nodes.map((node) => {
+                    return {
+                        properties: {
+                            [parentKey]: {
+                                ...getEnumValues([node]),
+                                ...getEnumNames([node]),
+                            },
+                            [childKey]: {
+                                ...getEnumValues(node.children),
+                                ...getEnumNames(node.children),
+                            },
+                        },
+                        ...buildDependencies(node.children),
+                    };
+                }),
+            },
+        },
+    };
+}
+
+/**
+ * Combine schema and dependencies block for usage with react-jsonschema-form (RJSF)
+ * @param {Object} schema - Schema
+ * @param {String} schemaId - Schema id (takes precedence over `schema` when both are provided)
+ * @param {Object[]} nodes - Array of nodes
+ * @param {Boolean} modifyProperties - Whether properties in main schema should be modified (add `enum` and `enumNames`)
+ * @returns {{}|{[p: string]: *}} - RJSF schema
+ */
+export function getSchemaWithDependencies({
+    schema = {},
+    schemaId,
+    nodes,
+    modifyProperties = false,
+}) {
+    const mainSchema = schemaId ? JSONSchemasInterface.schemaById(schemaId) : schema;
+
+    if (!isEmpty(mainSchema) && typeofSchema(mainSchema) !== "object") {
+        console.error("getSchemaWithDependencies() only accepts schemas of type 'object'");
+        return {};
+    }
+
+    // RJSF does not automatically render dropdown widget if `enum` is not present
+    if (modifyProperties && nodes.length) {
+        const mod = {
+            [nodes[0].dataSelector.key]: {
+                ...getEnumNames(nodes),
+                ...getEnumValues(nodes),
+            },
+        };
+        forEach(mod, (extraFields, key) => {
```

Comment:
Why not use the built-in Array.forEach?

Reply:
This is due to calling `forEach` on an object. I can also implement it using `Array.forEach` where `mod` is a list of objects.